### PR TITLE
libpng: cleanup for modules 3 -> 4 migration

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@teo, try to adapt your modulefiles as in this PR to see if modules4 digests it.

File: libpng.sh [hunk 1]
```diff
@@ -37,8 +37,7 @@ module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@
 # Dependencies
 module load BASE/1.0 $([[ $ALIEN_RUNTIME_VERSION ]] && echo "AliEn-Runtime/$ALIEN_RUNTIME_VERSION-$ALIEN_RUNTIME_REVISION" || echo "${ZLIB_VERSION:+zlib/$ZLIB_VERSION-$ZLIB_REVISION}")
 # Our environment
-setenv LIBPNG_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
-prepend-path PATH $::env(LIBPNG_ROOT)/bin
-prepend-path LD_LIBRARY_PATH $::env(LIBPNG_ROOT)/lib
-$([[ ${ARCHITECTURE:0:3} == osx ]] && echo "prepend-path DYLD_LIBRARY_PATH $::env(LIBPNG_ROOT)/lib")
+set LIBPNG_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
Comment (teo) [inline libpng.sh]: Just to confirm, we can keep using `::env` but only with `set` and never `setenv`?

Comment (ktf) [reply libpng.sh]: I don't know. :) I have two ideas on what might be wrong:

1) `append-path` does not allow for ":" anymore in 4 (at least as far as I can tell) and `$::env` should really become `$env` or something equivalent.
2) if you setenv, the variable is not available anymore within the same module file. 

I suspect the problem is 1.

Comment (teo) [reply libpng.sh]: I confirm that your change fixes this.

Comment (ktf) [reply libpng.sh]: Ok, then we have quite some cleanup to do... ;-)

**Pull request merged**